data_quality/tasks.py

```diff
@@ -337,7 +321,7 @@ def get_period_sources(self, period, sources):
         period_sources = []
 
         for source in sources:
-            if period == source['period_id']:
+            if period == source.get('period_id',''):
```

Comment:
If we have an empty value here, the whole database is broken, no? Therefore, allowing the code to continue with an empty value seems wrong.